cpp/src/arrow/util/compression_zstd.cc

```diff
@@ -180,12 +189,27 @@ Status ZSTDCompressor::End(int64_t output_len, uint8_t* output, int64_t* bytes_w
   *should_retry = ret > 0;
   return Status::OK();
 }
+} // namespace
 
 // ----------------------------------------------------------------------
 // ZSTD codec implementation
 
+ZSTDCodec::ZSTDCodec() : Codec(), zstdCompressionLevel_(kZSTDDefaultCompressionLevel) {
+  const char* envZstdCompressionLevel = getenv("ARROW_ZSTD_COMPRESSION_LEVEL");
```

Comment:
I think this is probably the wrong level to be reading environment variables (it seems like we potentially want this at different levels in the same binary), can it be pushed up the stack so it can perhaps be set explicitly.

Reply:
Also please use the `GetEnvVar` helper function from `arrow/util/io-util.h`.

Reply:
The use case seems to be Parquet file writing, btw, so perhaps the setting should be Parquet-specific?

Reply:
I would prefer having this setting added to the `parquet::ColumnProperties` object as that already manages related properties.  I don't really know enough about whether there can be one compression level or if there should be one specific for each codec.

Reply:
Done